### PR TITLE
test: add more integration test for new p2p lib

### PR DESCRIPTION
**Pull request by quake**
- [x] add network related rpc
- [x] test discovery
- [x] test disconnection

### CODE REVIEW

File: rpc/README.md [hunk 1]
```diff
@@ -302,19 +302,38 @@ curl -d '{"id": 2, "jsonrpc": "2.0", "method":"local_node_info","params": []}' -
...
+```shell
+curl -d '{"id": 2, "jsonrpc": "2.0", "method":"get_peers","params": []}' -H 'content-type:application/json' 'http://localhost:8114'
+```
+
+```json
+{
+    "jsonrpc": "2.0",
+    "result": [
+        {
+            "address": "/ip4/127.0.0.1/tcp/60470/p2p/QmdiJuQZj1dM4K4HKMxfMwcAqGFYvGKpbvVxTzyQeNGEcG",
```
Comment (jjyr) [inline rpc/README.md]: Should be an array.

File: rpc/README.md [hunk 2]
```diff
@@ -302,19 +302,38 @@ curl -d '{"id": 2, "jsonrpc": "2.0", "method":"local_node_info","params": []}' -
 {
     "jsonrpc": "2.0",
     "result": {
-        "addresses": [
-            {
-                "address": "/ip4/0.0.0.0/tcp/12344/p2p/QmWRU2NSro4wKgVbFX6y8SPFkcJ1tE2X5xzk9msMhdRmdS",
-                "score": 1
-            }
-        ],
+        "address": "/ip4/0.0.0.0/tcp/12344/p2p/QmWRU2NSro4wKgVbFX6y8SPFkcJ1tE2X5xzk9msMhdRmdS",
```
Comment (jjyr) [inline rpc/README.md]: Why change this?

Comment (quake) [reply rpc/README.md]: ops, my mistake, some copy/paste and rebase error... will resolve it later, thanks.

Comment (quake) [reply rpc/README.md]: @jjyr resolved, please review again, and for the new `get_peers` rpc, need your help to fix this issue:
https://github.com/nervosnetwork/ckb/blob/63fa068566c58997ad37c2edd46300c41d23c40a/rpc/src/module/net.rs#L49

We can submit a new fix after this PR and Identify refactoring PR are merged.

File: network/src/network_service.rs [hunk 3]
```diff
@@ -125,4 +126,14 @@ impl NetworkService {
...
     }
+
+    pub fn connected_peers(&self) -> Vec<(PeerId, Peer)> {
+        // TODO return listen addr for connected peers
+        self.network
+            .peers_registry
+            .read()
+            .peers_iter()
+            .map(|(peer_id, peer)| (peer_id.clone(), peer.clone()))
+            .collect()
```
Comment (jjyr) [inline network/src/network_service.rs]:
```
let peer_store = self.network.peer_store.read();
let (peer_id, peer, peer_addresses) = peers.map(|(peer_id, peer)| (peer_id.clone(), peer.clone(), peer_store.peer_addrs(peer_id, 3))).collect();
```

Comment (quake) [reply network/src/network_service.rs]: `peer_addrs` still returns incorrect port:

```json
{
  "jsonrpc": "2.0",
  "result": [
    {
      "addresses": [
        {
          "address": "/ip4/0.0.0.0/tcp/8117",
          "score": 1
        },
        {
          "address": "/ip4/127.0.0.1/tcp/55676/p2p/QmdiJuQZj1dM4K4HKMxfMwcAqGFYvGKpbvVxTzyQeNGEcG",
          "score": 1
        }
      ],
      "node_id": "QmdiJuQZj1dM4K4HKMxfMwcAqGFYvGKpbvVxTzyQeNGEcG",
      "version": "unknown"
    }
  ],
  "id": 2
}
```

I will leave this issue here, please help to add more assert in integration test after #351 is merged.

File: network/src/network_service.rs [hunk 4]
```diff
@@ -125,4 +126,27 @@ impl NetworkService {
...
+        self.network
+            .peers_registry
+            .read()
+            .peers_iter()
+            .map(|(peer_id, peer)| {
+                (
+                    peer_id.clone(),
+                    peer.clone(),
+                    peer_store
+                        .peer_addrs(peer_id, 3)
```
Comment (jjyr) [inline network/src/network_service.rs]: use a  constant variable

**Pull request merged**